lib/modules/manager/nuget/config-formatter.ts

```diff
@@ -0,0 +1,119 @@
+import * as hostRules from '../../../util/host-rules';
+import { NugetDatasource } from '../../datasource/nuget';
+import { parseRegistryUrl } from '../../datasource/nuget/common';
+import type { ParsedRegistryUrl } from '../../datasource/nuget/types';
+import type {
+  PackageSourceCredential,
+  PackageSourceMap,
+  Registry,
+} from './types';
+
+export function createNuGetConfigXml(registries: Registry[]): string {
+  let contents = `<?xml version="1.0" encoding="utf-8"?>\n<configuration>\n<packageSources>\n`;
+  let unnamedRegistryCount = 0;
+
+  const credentials: PackageSourceCredential[] = [];
+  const packageSourceMaps: PackageSourceMap[] = [];
+
+  for (const registry of registries) {
+    const registryName =
+      registry.name ?? `Package source ${++unnamedRegistryCount}`;
```

Comment:
I would prefer to sluggify the url and use it as name

Reply:
This copies the same behavior as the `dotnet nuget add source` command has, I figured it would be better not to break this behavior. What do you think?

Reply:
ok